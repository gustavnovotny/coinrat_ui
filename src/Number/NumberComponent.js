// @flow
import React from "react"
import {Colors} from "@blueprintjs/core"

type Props = {
  number: Number,
  colored: boolean,
  forceColor: ?string,
  percent: boolean,
  suffix: ?string,
}

function calculateColor(number: Number, forceColor: ?string): string {
  if (forceColor !== null) {
    return forceColor
  }
  if (number > 0) {
    return Colors.GREEN1
  } else if (number < 0) {
    return Colors.RED1
  }
}

const NumberComponent = ({number, colored = false, percent = false, forceColor = null, suffix = null}: Props) => {
  const color = colored ? calculateColor(number, forceColor) : Colors.BLACK
  const numberToDisplay = percent ? number * 100 : number
  const suffixToDisplay = suffix !== null ? suffix : (percent ? '%' : '')
  const precision = percent ? 2 : 8

  return <span className="pt-monospace-text" style={{color: color}}>
    {numberToDisplay.toFixed(precision)}{suffixToDisplay}
  </span>
}

export default NumberComponent
