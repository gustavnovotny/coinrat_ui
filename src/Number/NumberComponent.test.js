// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import NumberComponent from "./NumberComponent"

it('renders correctly - uncolored', () => {
  const tree = renderer
    .create(<NumberComponent number={1.24}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly - colored - positive', () => {
  const tree = renderer
    .create(<NumberComponent number={1.24} colored/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly - colored - negative', () => {
  const tree = renderer
    .create(<NumberComponent number={-1.24} colored/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
