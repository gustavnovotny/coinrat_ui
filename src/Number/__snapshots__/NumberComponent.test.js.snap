// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - colored - negative 1`] = `
<span
  className="pt-monospace-text"
  style={
    Object {
      "color": "#A82A2A",
    }
  }
>
  -1.24565986
   
  
</span>
`;

exports[`renders correctly - colored - positive 1`] = `
<span
  className="pt-monospace-text"
  style={
    Object {
      "color": "#0A6640",
    }
  }
>
  1.24565980
   
  
</span>
`;

exports[`renders correctly - percent 1`] = `
<span
  className="pt-monospace-text"
  style={
    Object {
      "color": "#A82A2A",
    }
  }
>
  -124.57
   
  %
</span>
`;

exports[`renders correctly - suffix, forceColor 1`] = `
<span
  className="pt-monospace-text"
  style={
    Object {
      "color": "#0E5A8A",
    }
  }
>
  -124.00
   
  yolo
</span>
`;

exports[`renders correctly - uncolored 1`] = `
<span
  className="pt-monospace-text"
  style={
    Object {
      "color": "#10161A",
    }
  }
>
  1.24565980
   
  
</span>
`;
